src/scripts/settings/webSettings.js

```diff
@@ -126,23 +111,29 @@ const checkDefaultTheme = (themes) => {
 
 export function getThemes() {
     return getConfig().then(config => {
-        const themes = Array.isArray(config.themes) ? config.themes : [];
+        if (!Array.isArray(config.themes)) {
+            console.error('web config is invalid, missing themes:', config);
+        }
+        const themes = Array.isArray(config.themes) ? config.themes : DefaultConfig.themes;
         checkDefaultTheme(themes);
         return themes;
     }).catch(error => {
         console.log('cannot get web config:', error);
         checkDefaultTheme();
-        return [];
+        return DefaultConfig.themes;
     });
```

Comment:
Do we even get into `catch` section?

Reply:
It shouldn't, but better to be on the safe side imo.